src/dotnet/Chat.UI.Blazor/Components/ChatView/MessageMenu/MessageMenuContent.razor

```diff
@@ -71,31 +76,36 @@
 }
 
 @code{
+    [Inject] private IChats Chats { get; set; } = null!;
     [Inject] private IChatAuthors ChatAuthors { get; set; } = null!;
     [Inject] private Session Session { get; set; } = null!;
     [Inject] private IEventAggregator EventAggregator { get; set; } = null!;
     [Inject] private NavigationManager Nav { get; set; } = null!;
+    [Inject] private ChatUI ChatUI { get; set; } = null!;
     [Inject] private ClipboardUI ClipboardUI { get; set; } = null!;
+    [Inject] private FeedbackUI FeedbackUI { get; set; } = null!;
 
-    private ChatAuthor RequestAuthor { get; set; } = null!;
-
-    [CascadingParameter] public ChatMessageModel Model { get; set; } = null!;
+    [CascadingParameter] public ChatMessageModel ChatMessageModel { get; set; } = null!;
 
-    private bool _isJoined;
-    private bool _isEntryAuthor;
-
-    protected override async Task OnInitializedAsync() {
-        await base.OnInitializedAsync();
-        await InitRequestAuthorId();
+    protected override ComputedState<Model>.Options GetStateOptions() {
+        return new ComputedState<Model>.Options() {
+            InitialValue = Model.None
+        };
     }
 
-    private async Task InitRequestAuthorId() {
-        var requestAuthor = await ChatAuthors.Get(Session, Model.Entry.ChatId, CancellationToken.None);
-        if (requestAuthor == null)
-            return;
-        RequestAuthor = requestAuthor;
-        _isEntryAuthor = RequestAuthor.Id == Model.Entry.AuthorId;
-        _isJoined = true;
+    protected override async Task<Model> ComputeState(CancellationToken cancellationToken)
+    {
+        var chatEntry = ChatMessageModel.Entry;
+        var chatId = chatEntry.ChatId;
+        var requestAuthor = await ChatAuthors.Get(Session, chatId, CancellationToken.None);
+        var rules = await Chats.GetRules(Session, chatId, cancellationToken).ConfigureAwait(false);
```

Comment:
this line contains ConfigureAwait(false), but the previous does not

Reply:
done